MethodParamPad ignores explicit constructor invocation

I have read check documentation: https://checkstyle.sourceforge.io/checks/whitespace/methodparampad.html#MethodParamPad
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash

➜  src cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <property name="severity" value="error"/>
  <property name="fileExtensions" value="java, properties, xml"/>
  <module name="TreeWalker">
    <module name="MethodParamPad">
      <property name="option" value="space"/>
    </module>
  </module>
</module>

➜  src cat -n Test.java
     1  public class Test {
     2      public Test () {} // ok, whitespace between id and ()
     3  
     4      public Test (int x) {
     5          this (); // ok, 'this' seems to be ignored
     6          toString (); // ok, whitespace between id and ()
     7      }
     8      public Test (int x, int y) {
     9          this(x + y); // should be a violation
    10          toString(); // violation
    11      }
    12  
    13      public Test (int x, int y, int z) {
    14          super(); // violation, super constructor call is checked
    15          toString(); // violation
    16      }
    17  
    18      public Test (int x, int y, int z, int a) {
    19          super (); // ok, whitespace between 'super' and ()
    20          toString(); // violation
    21      }
    22  }

➜  src javac Test.java 

➜  src java -jar checkstyle-10.12.3-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /home/nick/IdeaProjects/tester/src/Test.java:10:17: '(' is not preceded with whitespace. [MethodParamPad]
[ERROR] /home/nick/IdeaProjects/tester/src/Test.java:14:14: '(' is not preceded with whitespace. [MethodParamPad]
[ERROR] /home/nick/IdeaProjects/tester/src/Test.java:15:17: '(' is not preceded with whitespace. [MethodParamPad]
[ERROR] /home/nick/IdeaProjects/tester/src/Test.java:20:17: '(' is not preceded with whitespace. [MethodParamPad]
Audit done.
Checkstyle ends with 4 errors.

```

---

I would expect Checkstyle to show a violation for the following:
```java
this(x + y);
```

Example: 
```bash
➜  src java -jar checkstyle-10.12.3-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /home/nick/IdeaProjects/tester/src/Test.java:9:13: '(' is not preceded with whitespace. [MethodParamPad]
[ERROR] /home/nick/IdeaProjects/tester/src/Test.java:10:17: '(' is not preceded with whitespace. [MethodParamPad]
[ERROR] /home/nick/IdeaProjects/tester/src/Test.java:14:14: '(' is not preceded with whitespace. [MethodParamPad]
[ERROR] /home/nick/IdeaProjects/tester/src/Test.java:15:17: '(' is not preceded with whitespace. [MethodParamPad]
[ERROR] /home/nick/IdeaProjects/tester/src/Test.java:20:17: '(' is not preceded with whitespace. [MethodParamPad]
Audit done.
Checkstyle ends with 4 errors.

```

The TLDR; is that if this check validates super constructor invocations, it should also validate explicit constructor invocations.

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/MethodParamPadCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/MethodParamPadCheck.java
@@ -59,6 +59,8 @@ import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
  * Default value is:
  * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#CTOR_DEF">
  * CTOR_DEF</a>,
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#CTOR_CALL">
+ * CTOR_CALL</a>,
  * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_NEW">
  * LITERAL_NEW</a>,
  * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#METHOD_CALL">
@@ -133,6 +135,7 @@ public class MethodParamPadCheck
     public int[] getAcceptableTokens() {
         return new int[] {
             TokenTypes.CTOR_DEF,
+            TokenTypes.CTOR_CALL,
             TokenTypes.LITERAL_NEW,
             TokenTypes.METHOD_CALL,
             TokenTypes.METHOD_DEF,

--- a/src/main/resources/com/puppycrawl/tools/checkstyle/meta/checks/whitespace/MethodParamPadCheck.xml
+++ b/src/main/resources/com/puppycrawl/tools/checkstyle/meta/checks/whitespace/MethodParamPadCheck.xml
@@ -25,7 +25,7 @@
                       type="com.puppycrawl.tools.checkstyle.checks.whitespace.PadOption">
                <description>Specify policy on how to pad method parameter.</description>
             </property>
-            <property default-value="CTOR_DEF,LITERAL_NEW,METHOD_CALL,METHOD_DEF,SUPER_CTOR_CALL,ENUM_CONSTANT_DEF,RECORD_DEF"
+            <property default-value="CTOR_DEF,CTOR_CALL,LITERAL_NEW,METHOD_CALL,METHOD_DEF,SUPER_CTOR_CALL,ENUM_CONSTANT_DEF,RECORD_DEF"
                       name="tokens"
                       type="java.lang.String[]"
                       validation-type="tokenSet">

--- a/src/main/resources/google_checks.xml
+++ b/src/main/resources/google_checks.xml
@@ -277,7 +277,7 @@
     </module>
     <module name="MethodParamPad">
       <property name="tokens"
-               value="CTOR_DEF, LITERAL_NEW, METHOD_CALL, METHOD_DEF,
+               value="CTOR_DEF, LITERAL_NEW, METHOD_CALL, METHOD_DEF, CTOR_CALL,
                     SUPER_CTOR_CALL, ENUM_CONSTANT_DEF, RECORD_DEF"/>
     </module>
     <module name="NoWhitespaceBefore">

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/MethodParamPadCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/MethodParamPadCheckTest.java
@@ -128,6 +128,7 @@ public class MethodParamPadCheckTest
             "20:34: " + getCheckMessage(MSG_WS_PRECEDED, "("),
             "31:17: " + getCheckMessage(MSG_WS_PRECEDED, "("),
             "32:14: " + getCheckMessage(MSG_WS_PRECEDED, "("),
+            "33:18: " + getCheckMessage(MSG_WS_PRECEDED, "("),
             "37:17: " + getCheckMessage(MSG_WS_PRECEDED, "("),
             "38:33: " + getCheckMessage(MSG_WS_PRECEDED, "("),
             "44:17: " + getCheckMessage(MSG_WS_PRECEDED, "("),
@@ -141,7 +142,9 @@ public class MethodParamPadCheckTest
 
     @Test
     public void test1322879() throws Exception {
-        final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
+        final String[] expected = {
+            "28:13: " + getCheckMessage(MSG_WS_NOT_PRECEDED, "("),
+        };
         verifyWithInlineConfigParser(
                 getPath("InputMethodParamPadWhitespaceAround.java"),
                expected);
@@ -165,11 +168,23 @@ public class MethodParamPadCheckTest
     }
 
     @Test
+    public void testMethodParamPadCheckConstructors() throws Exception {
+        final String[] expected = {
+            "21:15: " + getCheckMessage(MSG_WS_PRECEDED, "("),
+            "25:14: " + getCheckMessage(MSG_WS_PRECEDED, "("),
+            "29:14: " + getCheckMessage(MSG_WS_PRECEDED, "("),
+        };
+        verifyWithInlineConfigParser(
+                getPath("InputMethodParamPadCheckConstructors.java"), expected);
+    }
+
+    @Test
     public void testGetAcceptableTokens() {
         final MethodParamPadCheck methodParamPadCheckObj = new MethodParamPadCheck();
         final int[] actual = methodParamPadCheckObj.getAcceptableTokens();
         final int[] expected = {
             TokenTypes.CTOR_DEF,
+            TokenTypes.CTOR_CALL,
             TokenTypes.LITERAL_NEW,
             TokenTypes.METHOD_CALL,
             TokenTypes.METHOD_DEF,

--- a/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/whitespace/methodparampad/InputMethodParamPadRecords.java
+++ b/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/whitespace/methodparampad/InputMethodParamPadRecords.java
@@ -30,7 +30,7 @@ public class InputMethodParamPadRecords {
     //in ctor
     record Mtr2 () { // violation ''(' is preceded with whitespace'
         Mtr2 (String s1, String s2, String s3){ // violation ''(' is preceded with whitespace'
-            this ();
+            this (); // violation ''(' is preceded with whitespace'
         }
     }
 

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/methodparampad/InputMethodParamPadCheckConstructors.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/methodparampad/InputMethodParamPadCheckConstructors.java
@@ -1,0 +1,40 @@
+/*
+MethodParamPad
+allowLineBreaks = true
+option = (default)nospace
+tokens = (default)CTOR_DEF, LITERAL_NEW, METHOD_CALL, METHOD_DEF, SUPER_CTOR_CALL, \
+         CTOR_CALL, ENUM_CONSTANT_DEF, RECORD_DEF
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.whitespace.methodparampad;
+
+import java.util.ArrayList;
+
+public class InputMethodParamPadCheckConstructors extends ArrayList {
+    public InputMethodParamPadCheckConstructors() {
+        super(); // ok, no whitespace
+    }
+
+    public InputMethodParamPadCheckConstructors(Object b) {
+        super (10); // violation ''(' is preceded with whitespace.'
+    }
+
+    public InputMethodParamPadCheckConstructors(int a) {
+        this (); // violation ''(' is preceded with whitespace.'
+    }
+
+    public InputMethodParamPadCheckConstructors(int a, String b) {
+        this (a + b.length()); // violation ''(' is preceded with whitespace.'
+    }
+
+    public InputMethodParamPadCheckConstructors(String a, String b) {
+        this
+            (a.length() + b.length()); // line break is ok, no violation
+    }
+
+    public InputMethodParamPadCheckConstructors(String a, String b, Object c) {
+        this(a.length() + b.length()); // ok, no whitespace
+    }
+}

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/methodparampad/InputMethodParamPadWhitespaceAround.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/methodparampad/InputMethodParamPadWhitespaceAround.java
@@ -25,8 +25,7 @@ public class InputMethodParamPadWhitespaceAround
 
     protected InputMethodParamPadWhitespaceAround ( String s)
     {
-        // ok, until https://github.com/checkstyle/checkstyle/issues/13675
-        this();
+        this(); // violation ''(' is not preceded with whitespace'
     }
 
     public void enhancedFor ()

--- a/src/xdocs/checks/whitespace/methodparampad.xml
+++ b/src/xdocs/checks/whitespace/methodparampad.xml
@@ -52,6 +52,8 @@
               <td>subset of tokens
                 <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#CTOR_DEF">
                     CTOR_DEF</a>
+                , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#CTOR_CALL">
+                    CTOR_CALL</a>
                 , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_NEW">
                     LITERAL_NEW</a>
                 , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#METHOD_CALL">
@@ -69,6 +71,8 @@
               <td>
                 <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#CTOR_DEF">
                     CTOR_DEF</a>
+                , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#CTOR_CALL">
+                    CTOR_CALL</a>
                 , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_NEW">
                     LITERAL_NEW</a>
                 , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#METHOD_CALL">

